SimulationIOFilters/ImportDelamData.cpp

```diff
@@ -0,0 +1,568 @@
+    return;
+  }
+
+  if(!fi.isFile() || fi.suffix() != "txt")
+  {
+    QString ss = QObject::tr("Bvid File is not a text file.  Please enter the path to a text file.");
+    setErrorCondition(-2001, ss);
+    return;
+  }
+
+  fi.setFile(m_BvidStdOutFile);
+  if(m_BvidStdOutFile.isEmpty())
+  {
+    QString ss = QObject::tr("Bvid StdOut File is empty.  Please enter a file path.");
+    setErrorCondition(-2002, ss);
+    return;
+  }
+
+  if(!fi.isFile() || fi.completeSuffix() != "txt")
+  {
+    QString ss = QObject::tr("Bvid StdOut File is not a text file.  Please enter the path to a text file.");
+    setErrorCondition(-2003, ss);
+    return;
+  }
+
+  readBvidStdOutFile();
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  DataContainerShPtr dc = getDataContainerArray()->createNonPrereqDataContainer(this, getDataContainerName(), DataContainerID);
+
+  m_Xcrd = static_cast<int>(*m_Lcx / *m_Ex) + 1;
+  m_Ycrd = m_Xcrd;
+  m_Zcrd = 2 * *m_CzmLayers + 2;
+
+  float translate_x = *m_Ex * static_cast<int>(*m_Lcx / *m_Ex / 2);
+  float translate_y = translate_x;
+
+  FloatArrayType::Pointer xBounds = FloatArrayType::CreateArray(m_Xcrd, SIMPL::Geometry::xBoundsList, true);
+  std::generate(xBounds->begin(), xBounds->end(), [n = -translate_x, this]() mutable { return n += *m_Ex; });
+
+  FloatArrayType::Pointer yBounds = FloatArrayType::CreateArray(m_Ycrd, SIMPL::Geometry::yBoundsList, true);
+  std::generate(yBounds->begin(), yBounds->end(), [n = -translate_y, this]() mutable { return n += *m_Ex; });
+
+  FloatArrayType::Pointer zBounds = FloatArrayType::CreateArray(m_Zcrd, SIMPL::Geometry::zBoundsList, true);
+  float z_0 = 0.0f;
+  float z_1 = (*m_TotalThk / *m_NumPlies) - m_InterfaceThickness;
+  zBounds->setValue(0, z_0);
+  zBounds->setValue(1, z_1);
+
+  int currentIdx = 2;
+  for(int i = 1; i <= *m_NumPlies - 1; i++)
+  {
+    float z_0 = (i * *m_TotalThk / *m_NumPlies) + m_InterfaceThickness;
+    float z_1 = ((i + 1) * *m_TotalThk / *m_NumPlies) - m_InterfaceThickness;
+    zBounds->setValue(currentIdx++, z_0);
+    zBounds->setValue(currentIdx++, z_1);
+  }
+
+  RectGridGeom::Pointer geom = RectGridGeom::CreateGeometry(SIMPL::Geometry::RectGridGeometry);
+  geom->setDimensions({m_Xcrd - 1, m_Ycrd - 1, m_Zcrd - 1});
+  geom->setXBounds(xBounds);
+  geom->setYBounds(yBounds);
+  geom->setZBounds(zBounds);
+  dc->setGeometry(geom);
+
+  size_t totalLines = getBvidFileLineCount();
+  if(totalLines == 0)
+  {
+    return;
+  }
+
+  AttributeMatrixShPtr am =
+      dc->createNonPrereqAttributeMatrix(this, getCellAttributeMatrixName(), geom->getDimensions().toContainer<std::vector<size_t>>(), AttributeMatrix::Type::Cell, AttributeMatrixID21);
+  am->createNonPrereqArray<FloatArrayType>(this, getDataArrayName(), 0, {1}, DataArrayID30);
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::readBvidStdOutFile()
+{
+  QFile bvidStdOutFile(m_BvidStdOutFile);
```

Comment:
```suggestion
  m_Lcx.reset();
  m_Ex.reset();
  m_CzmLayers.reset();
  m_NumPlies.reset();
  m_TotalThk.reset();
  QFile bvidStdOutFile(m_BvidStdOutFile);
```

I think the std::optional variables that get read here need to be reset. For example if you change file, the old values will still be present. And if the new file is missing one of the variables, the old value will be mistakenly be used.